src/ngircd/irc-channel.c

```diff
@@ -236,8 +237,56 @@ join_forward(CLIENT *Client, CLIENT *target, CHANNEL *chan,
 	}
 
 	/* tell users in this channel about the new client */
+		#define IMPLEMENT_EXTENDED_JOIN
+	#ifndef IMPLEMENT_EXTENDED_JOIN
 	IRC_WriteStrChannelPrefix(Client, chan, target, false,
 				  "JOIN :%s",  channame);
+	#endif
+	#ifdef IMPLEMENT_EXTENDED_JOIN
+	CLIENT *c;
+	// Pointless alias
```

Comment:
C++ comments are not supported by old compilers. Please stick to C style comments: `/* … */`.
And what do you mean with this comment at all?